pkg/pattern/drain/metrics.go

```diff
@@ -1,8 +1,33 @@
 package drain
 
-import "github.com/prometheus/client_golang/prometheus"
+import (
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	FormatLogfmt  = "logfmt"
+	FormatJson    = "json"
+	FormatUnknown = "unknown"
+)
+
+// DetectLogFormat guesses at how the logs are encoded based on some simple heuristics.
+// It only runs on the first log line when a new stream is created, so it could do some more complex parsing or regex.
+func DetectLogFormat(line string) string {
+	if len(line) < 2 {
+		return FormatUnknown
+	} else if line[0] == '{' && line[len(line)-1] == '}' {
+		return FormatJson
+	} else if strings.Count(line, "=") > strings.Count(line, " ")-5 {
```

Comment:
why -5 ?

Reply:
Honestly, that number is what worked against our testdata! I think this is *slightly* better than only checking if the string contains an `=`.
If you feel strongly about making this more robust, I can rewrite it as a regex. Its only used once per stream so that shouldn't be a problem, performance wise.

Reply:
I think we need something a bit more robust. There could be a lot of spaces in multiple fields ?